src/storage/object/delete.js

```diff
@@ -31,32 +33,41 @@ export async function deleteObject(env, daCtx, key, isMove = false) {
   await env.DA_CONTENT.delete(`${daCtx.org}/${key}`);
   await deleteFromCollab(env, tmpCtx);
 }
+
 /**
  * Deletes one or more objects in the storage. Object is specified by the key in the daCtx or a list passed in.
  * Note: folders can not be specified in the `keys` list.
  *
  * @param {Object} env the CloudFlare environment
  * @param {DaCtx} daCtx the DA Context
- * @return {Promise<{body: null, status: number}>}
+ * @param {Object} details the details about the delete operation
+ * @return {Promise<{body: {}, status: number}>}
  */
-export default async function deleteObjects(env, daCtx) {
+export default async function deleteObjects(env, daCtx, details) {
+  if (daCtx.isFile) {
+    await deleteObject(env, daCtx, daCtx.key);
+    return { body: null, status: 204 };
+  }
+
   const keys = [];
+  if (!daCtx.isFile && !details.continuationToken) {
+    keys.push(`${daCtx.key}.props`);
```

Comment:
And why do we need to add the directory (.props) if the continuationToken isn't set?

Reply:
We need to add the directory sidecar file (`.props`) to the list of files to delete, in the case of a delete request for a directory.